cmd/discovery/main.go

```diff
@@ -1,53 +1,670 @@
+	// Global/persistent flags
+	debug bool
+)
+
+// rootCmd represents the base command (TPM hash generation)
+var rootCmd = &cobra.Command{
+	Use:   "kcrypt-discovery-challenger",
+	Short: "kcrypt-challenger discovery client",
+	Long: `kcrypt-challenger discovery client
+
+This tool provides TPM-based operations for encrypted partition management.
+By default, it outputs the TPM hash for this device.
+
+Configuration:
+  The client reads configuration from Kairos configuration files in the following directories:
+  - /oem (during installation from ISO)
+  - /sysroot/oem (on installed systems during initramfs)
+  - /tmp/oem (when running in hooks)
+
+  Configuration format (YAML):
+    kcrypt:
+      challenger:
+        challenger_server: "https://my-server.com:8082"  # Server URL
+        mdns: true                                       # Enable mDNS discovery
+        certificate: "/path/to/server-cert.pem"         # Server certificate
+        nv_index: "0x1500000"                           # TPM NV index (offline mode)
+        c_index: "0x1500001"                            # TPM certificate index
+        tpm_device: "/dev/tpmrm0"                        # TPM device path`,
+	Example: `  # Get TPM hash for this device (default)
+  kcrypt-discovery-challenger
+
+  # Get passphrase for encrypted partition
+  kcrypt-discovery-challenger get --partition-name=/dev/sda2
+
+  # Display TPM enrollment information (hash, PCRs, EK)
+  kcrypt-discovery-challenger info
+
+  # Clean up TPM NV memory (useful for development)
+  kcrypt-discovery-challenger cleanup
+
+  # Run plugin event
+  kcrypt-discovery-challenger discovery.password`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return runTPMHash()
+	},
+}
+
+// newInfoCmd creates the info command
+func newInfoCmd() *cobra.Command {
+	var pcrs string
+
+	cmd := &cobra.Command{
+		Use:   "info",
+		Short: "Display TPM enrollment information",
+		Long: `Display TPM information used for remote KMS enrollment.
+
+This command shows the TPM hash, PCR values, and EK public key that are
+sent to the remote KMS during enrollment. This is useful for debugging
+and comparing client-side values with what's stored on the server.
+
+The --pcrs flag accepts a comma-separated list of PCR indices to display.
+If not specified, defaults to PCRs 0,7,11 (the standard set used for enrollment).`,
+		Example: `  # Display info with default PCRs (0,7,11)
+  kcrypt-discovery-challenger info
+
+  # Display info with specific PCRs
+  kcrypt-discovery-challenger info --pcrs=0,1,2,7
+
+  # Display info with all PCRs
+  kcrypt-discovery-challenger info --pcrs=0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runInfo(pcrs)
+		},
+	}
+
+	cmd.Flags().StringVar(&pcrs, "pcrs", "0,7,11", "Comma-separated list of PCR indices to display")
+
+	return cmd
+}
+
+// newCleanupCmd creates the cleanup command
+func newCleanupCmd() *cobra.Command {
```

Comment:
I think this now belongs to the kairos-agent, because it's only relevant to local passphrase logic. We can completely remove this but I find it a useful helper for people that want to delete the passphrase from the tpm (e.g. repurposing hardware and they don't want to expose the passphrase to the next user of the device).

Reply:
I will move it here: https://github.com/kairos-io/kairos-agent/pull/1027